analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd

```diff
@@ -0,0 +1,227 @@
+```{r}
+# File from 00-LGAT-select-pathology-dx that is used for the pathology diagnosis
+# inclusion/exclusion criteria
+path_dx_list <- jsonlite::fromJSON(
+  file.path(subset_dir, 
+            "lgat_subtyping_path_dx_strings.json")
+)
+
+# clinical file
+clinical <- read_tsv(file.path(root_dir, 
+                              "data",
+                              "pbta-histologies.tsv"), 
+                    guess_max = 10000)
+
+# consensus mutation data
+consensusMutation <- read_tsv(file.path(root_dir,
+                                        "data",
+                                        "pbta-snv-consensus-mutation.maf.tsv.gz"))
+
+# combined list for SNV of interest per subtype
+# additional information for genes with known hotspots/canonical mutation 
+# is also provided
+snvOI <- jsonlite::fromJSON(file.path(root_dir,
+                                      "analyses",
+                                      "molecular-subtyping-LGAT",
+                                      "input",
+                                      "snvOI_list.json"))
+
+```
+
+Collase multiple hotspots in genes with "|" so easy grep calls
+
+```{r}
+BRAF_hotspot <-paste(snvOI$BRAF_V600E$hotspot[!is.na( snvOI$BRAF_V600E$hotspot)],collapse = "|")
+FGFR_hotspot <-paste(snvOI$FGFR$hotspot[!is.na( snvOI$FGFR$hotspot)],collapse = "|")
+IDH_hotspot<- paste(snvOI$IDH$hotspot[!is.na( snvOI$IDH$hotspot)],collapse = "|")
+```
+
+Filter to tumor samples that should be included on the basis of pathology diagnosis
+```{r}
+lgat_specimens_df <- clinical %>%
+  dplyr::filter(str_detect(str_to_lower(pathology_diagnosis),  # Inclusion criteria
+                    paste0(path_dx_list$include_path_dx, collapse = "|")),
+         # Exclusion criteria
+         str_detect(str_to_lower(pathology_diagnosis),
+                    paste0(path_dx_list$exclude_path_dx, collapse = "|"),
+                    negate = TRUE),
+         # Tumors
+         sample_type == "Tumor",
+         composition == "Solid Tissue")
+
+# Write this intermediate file to the subset directory as it allows for
+# inspection
+write_tsv(lgat_specimens_df, file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to WGS samples
+lgat_wgs_df <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::select(Kids_First_Biospecimen_ID)
+```
+
+### Subset consensus maf and annotate per subtype based on SNV
+
+```{r}
+# Filter consensus mutation files for LGAT subset
+consensusMutationSubset <- consensusMutation %>%
+  # find lgat samples
+  dplyr::filter(Tumor_Sample_Barcode %in% lgat_wgs_df$Kids_First_Biospecimen_ID) %>%
+  # select tumor sample barcode
+  dplyr::select(Tumor_Sample_Barcode,
+                Hugo_Symbol,
+                HGVSp_Short,
+                DOMAINS,
+                Variant_Classification) %>% 
+  dplyr::distinct() %>%
+  # join other WGS LGAT samples
+  dplyr::full_join(lgat_wgs_df,
+            by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")) %>%
```

Comment:
Since I changed that to pull, this is no longer a df, but do you need this? Is the purpose here to determine if there are any samples which do not have mutations and to include those with NAs in the file? I don't see any cases of these. Perhaps you can do a setdiff to determine which do not have mutations so they are printed out, but when you do your later searching to add to the final file, I don't think they would be needed.

Reply:
you are right this is not required at this step.

I'll add a setdiff to identify why we need to add all LGAT WGS in step`Gather status of all SNVs per biospecimen` to gather all LGAT biospecimens.